### ISSUE TITLE
cimport does not work for functions returning tuples

### ISSUE THREAD

**Issue by MaxNoe** [cython/cython#2745]
With the following structure:

```
test/
    __init__.py
    test.pyx
    lib.pyx
    lib.pxd
setup.py
```

test.pyx:
```
# cython: language_level=3
from .lib cimport foo

print(foo())
```

lib.pyx
```
# cython: language_level=3
cdef (double, double) foo():
    return 5.0, 3.0
```

lib.pxd
```
# cython: language_level=3
cdef (double, double) foo()
```

setup.py
```
from setuptools import setup, find_packages
from Cython.Build import cythonize

setup(
    name='test',
    packages=find_packages(),
    ext_modules=cythonize('test/test.pyx'),
)
```

Running `python setup.py build_ext` raises the following compiler error:
```
test/test.c:1002:1: warning: parameter names (without types) in function declaration
 static PyObject* __pyx_convert__to_py___pyx_ctuple_double__and_double(__pyx_ctuple_double__and_double);
 ^~~~~~
test/test.c:1037:41: error: expected declaration specifiers or ‘...’ before ‘*’ token
 static __pyx_ctuple_double__and_double (*__pyx_f_4test_3lib_foo)(void); /*proto*/
                                         ^
test/test.c: In function ‘__Pyx_modinit_function_import_code’:
test/test.c:1194:64: error: ‘__pyx_f_4test_3lib_foo’ undeclared (first use in this function); did you mean ‘__pyx_k_test’?
   if (__Pyx_ImportFunction(__pyx_t_1, "foo", (void (**)(void))&__pyx_f_4test_3lib_foo, "__pyx_ctuple_double__and_double (void)") < 0) __PYX_ERR(0, 1, __pyx_L1_error)
                                                                ^~~~~~~~~~~~~~~~~~~~~~
                                                                __pyx_k_test
test/test.c:1194:64: note: each undeclared identifier is reported only once for each function it appears in
test/test.c: In function ‘__pyx_pymod_exec_test’:
test/test.c:1406:68: warning: implicit declaration of function ‘__pyx_f_4test_3lib_foo’ [-Wimplicit-function-declaration]
   __pyx_t_1 = __pyx_convert__to_py___pyx_ctuple_double__and_double(__pyx_f_4test_3lib_foo()); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 4, __pyx_L1_error)
                                                                    ^~~~~~~~~~~~~~~~~~~~~~
test/test.c: At top level:
test/test.c:1948:71: error: unknown type name ‘__pyx_ctuple_double__and_double’
 static PyObject* __pyx_convert__to_py___pyx_ctuple_double__and_double(__pyx_ctuple_double__and_double value) {
                                                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
test/test.c:1002:18: warning: ‘__pyx_convert__to_py___pyx_ctuple_double__and_double’ used but never defined
 static PyObject* __pyx_convert__to_py___pyx_ctuple_double__and_double(__pyx_ctuple_double__and_double);
                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
error: command 'gcc' failed with exit status 1
```

**Comment by scoder** [comment]
Thanks for the report. It seems that we're not declaring the return type of the cimported function, at least not in a way that would trigger the necessary ctuple machinery. Definitely a bug. Might be easy to fix by making sure that cimported types are looked at when seting up the ctuples, assuming that's really the problem.

**Comment by Baekalfen** [comment]
I can confirm this issue. I had to add a `cdef (int, int) _dummy_declaration` to the `.pxd` files of Python scripts, which used a `(int, int)` ctuple declaration through a called function.

**Comment by scoder** [comment]
Duplicate of #1427.
